src/clients/victoriaMetrics/promAgent.ts

```diff
@@ -0,0 +1,96 @@
+import promClient from 'prom-client';
+import { sendToDiscordAdmins } from '../discord';
+
+const MAX_PUSH_METRICS_ERRORS = 5;
+const FORGET_ERROR_TIME_THRESHOLD = 60 * 60 * 1000; // 1 hour
+
+export class PromAgent {
+  private metricNames = new Set<string>();
+  private pushMetricsErrorsTimestamps: number[] = [];
+  private readonly dupeMetricErrorHandler: (metricName: string) => void;
+
+  constructor(dupeMetricErrorHandler: (metricName: string) => void) {
+    this.dupeMetricErrorHandler = dupeMetricErrorHandler;
+  }
+
+  public registerMetric(metricName: string) {
+    this.metricNames.has(metricName)
+      ? this.dupeMetricErrorHandler(metricName)
+      : this.metricNames.add(metricName);
+  }
+
+  public async pushMetrics() {
+    let metrics: string;
+
+    try {
+      metrics = await promClient.register.metrics(); // Will call any collect() functions for gauges
+    } catch (e) {
+      // Exit program if non-initialised labels are used in collect() function
+      if (e.message.includes('label')) {
+        console.error(e);
+        process.exit(1);
+      }
+
+      throw e;
+    }
+
+    const response = await fetch(process.env.VM_IMPORT_PROMETHEUS_URL, {
```

Comment:
Can u test what happens if u don't include this in ur .env

`process.env.VM_IMPORT_PROMETHEUS_URL`

Reply:
Maybe just above it here, do a

```
if (process.env.VM_IMPORT_PROMETHEUS_URL === undefined) {
    process.exit(1);
}
```